### PR TITLE
add account to other accounts to internal transfers get detected

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Please write a summary of your changes and why you made them. 

fixes issue https://github.com/iotaledger/wallet.rs/issues/510

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Created an account in Firefly, received 10 Mi to the first address
Logged out
Logged in again, created a second account
Sent an internal transfers, it showed up as internal transfer

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/account/mod.rs [hunk 1]
```diff
@@ -355,6 +355,9 @@ impl AccountInitialiser {
                 self.sync_accounts_lock.clone(),
             );
             drop(accounts);
+            for (_, account) in self.accounts.read().await.iter(){
+                account.accounts.write().await.insert(account_id.clone(), guard.clone());
```
Comment (lucasfernog) [inline src/account/mod.rs:359]: `account.accounts` should be the same reference everywhere. `account.accounts == self.accounts` i's supposed to be true. but if this is just a workaround, pls put a note above it so we can find the actual issue later.

File: src/account/mod.rs [hunk 2]
```diff
@@ -355,6 +355,9 @@ impl AccountInitialiser {
                 self.sync_accounts_lock.clone(),
             );
             drop(accounts);
+            for (_, account) in self.accounts.read().await.iter(){
```
Comment (lucasfernog) [inline src/account/mod.rs:358]:
```suggestion
            for account in self.accounts.read().await.values() {
```

**Comment by lucasfernog** [comment]
Also this should be based on `develop` branch instead of `migration`.

**Comment by Thoralf-M** [comment]
Moved to this PR https://github.com/iotaledger/wallet.rs/pull/549

**Pull request closed without merging**